VisualPinball.Unity/VisualPinball.Unity.Editor/Layers/LayerTreeElement.cs

```diff
@@ -0,0 +1,114 @@
+﻿using UnityEditor;
+using UnityEngine;
+using VisualPinball.Engine.VPT.Table;
+using VisualPinball.Unity.Editor.Utils.TreeViewWithTreeModel;
+using VisualPinball.Unity.VPT;
+using VisualPinball.Unity.VPT.Table;
+
+namespace VisualPinball.Unity.Editor.Layers
+{
+	/// <summary>
+	/// Enum for different Item Types in the LayerTreeView
+	/// </summary>
+	/// <remarks>
+	/// The Item Type will change the IsVisible behavior.
+	/// It will be used also to display a specific icon per item in the treeview
+	/// </remarks>
+	enum LayerTreeViewElementType
+	{
+		Root,
+		Table,
+		Layer,
+		Item
+	}
+
+	/// <summary>
+	/// Element type used in TreeModel for the LayerTreeView
+	/// </summary>
+	/// <remarks>
+	/// Could be initialized in different ways to fit all possible LayerTreeView item types
+	/// </remarks>
+	class LayerTreeElement : TreeElement
+	{
+		public Table Table { get; private set; }
+		public ILayerableItemBehavior Item { get; private set; }
+		public string LayerName { get; private set; }
+		public LayerTreeViewElementType Type { get { return Table != null ? LayerTreeViewElementType.Table : Item != null ? LayerTreeViewElementType.Item : LayerName != null ? LayerTreeViewElementType.Layer : LayerTreeViewElementType.Root; } }
+
+		[SerializeField]

```

Comment:
Rider tells me this is redundant. Maybe all members of a `TreeElement` are serialized?

Reply:
It was mainly in case of saving TreeModel to Asset afaik